lib/dry/schema/dsl.rb

```diff
@@ -316,7 +323,7 @@ def new(**options, &block)
       # @api private
       def set_type(name, spec)
         type = resolve_type(spec)
-        meta = {required: false, maybe: type.optional?}
```

Comment:
By setting `required` to `true` here, we can be assured that each type will be able to be used correctly for any downstream value coercion. The value coercion's use of `type_schema` applies `lax` to the underlying types. Without this change, all types are somewhat lax by default which, in the case of operator composition, leads to incorrect removal of keys, as the semi-lax types will cause non-matching type cases to succeed.

Since none of the tests failed with this change, I feel pretty safe about this, but are there some cases where this might cause issues?